src/pages/iou/request/step/IOURequestStepMerchant.js

```diff
@@ -0,0 +1,126 @@
+import React, {useCallback, useRef} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import Form from '@components/Form';
+import TextInput from '@components/TextInput';
+import transactionPropTypes from '@components/transactionPropTypes';
+import useLocalize from '@hooks/useLocalize';
+import Navigation from '@libs/Navigation/Navigation';
+import styles from '@styles/styles';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import StepScreenWrapper from './StepScreenWrapper';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /** Onyx Props */
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+function IOURequestStepMerchant({
+    route: {
+        params: {iouType, reportID, transactionID},
+    },
+    transaction: {merchant},
+}) {
+    const {translate} = useLocalize();
+    const inputRef = useRef(null);
+    const focusTimeoutRef = useRef(null);
+
+    useFocusEffect(
+        useCallback(() => {
+            focusTimeoutRef.current = setTimeout(() => {
+                if (inputRef.current) {
+                    inputRef.current.focus();
+                }
+                return () => {
+                    if (!focusTimeoutRef.current) {
+                        return;
+                    }
+                    clearTimeout(focusTimeoutRef.current);
+                };
+            }, CONST.ANIMATED_TRANSITION);
+        }, []),
+    );
+
+    const navigateBack = () => {
+        Navigation.goBack(ROUTES.MONEYTEMPORARYFORREFACTOR_REQUEST_STEP.getRoute(iouType, CONST.IOU.REQUEST_STEPS.CONFIRMATION, transactionID, reportID), true);
+    };
+
+    /**
+     * @param {Object} value
+     * @param {String} value.moneyRequestMerchant
+     */
+    const validate = useCallback((value) => {
+        const errors = {};
+
+        if (_.isEmpty(value.moneyRequestMerchant)) {
+            errors.moneyRequestMerchant = 'common.error.fieldRequired';
+        }
+
+        return errors;
+    }, []);
+
+    /**
+     * @param {Object} value
+     * @param {String} value.moneyRequestMerchant
+     */
+    const updateMerchant = (value) => {
+        IOU.setMoneyRequestMerchant_temporaryForRefactor(transactionID, value.moneyRequestMerchant);
+        navigateBack();
+    };
+
+    return (
+        <StepScreenWrapper
+            headerTitle={translate('common.merchant')}
+            onBackButtonPress={navigateBack}
+            shouldShowWrapper
+            testID={IOURequestStepMerchant.displayName}
+        >
+            <Form
```

Comment:
We should be using the new FormProvider and InputWrapper components, same goes for the other pages using Form